Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -113,12 +123,24 @@ class TIFFLoadingContext {
             return {};
         };
 
+        auto read_alpha = [&]() -> ErrorOr<u8> {
+            if (m_metadata.extra_samples().has_value()) {
+                auto const extra_sample = m_metadata.extra_samples().value();
+                if (extra_sample.size() > 0 && extra_sample[extra_sample.size() - 1] == ExtraSample::AssociatedAlpha) {
+                    return TRY(read_component(stream, bits_per_sample[3 + extra_sample.size() - 1]));
+                }
+            }
+            return NumericLimits<u8>::max();
+        };
+
         if (m_metadata.photometric_interpretation() == PhotometricInterpretation::RGB) {
             auto const first_component = TRY(read_component(stream, bits_per_sample[0]));
             auto const second_component = TRY(read_component(stream, bits_per_sample[1]));
             auto const third_component = TRY(read_component(stream, bits_per_sample[2]));
             TRY(discard_unknown_channels());
-            return Color(first_component, second_component, third_component);
+
+            auto const alpha = TRY(read_alpha());
```

Comment:
does the spec say that alpha is after unknown channels? (if so, add spec comment with the quote)